inject-java/src/main/java/io/micronaut/annotation/processing/visitor/JavaClassElement.java

```diff
@@ -796,15 +801,32 @@ protected Collection<TypeElement> getInterfaces(TypeElement classNode) {
         }
 
         @Override
-        protected List<Element> getEnclosedElements(TypeElement classNode, ElementQuery.Result<?> result) {
+        protected List<Element> getEnclosedElements(TypeElement classNode, ElementQuery.Result<?> result, boolean includeAbstract) {
             List<? extends Element> ee;
             if (classNode == classElement) {
                 ee = getEnclosedElements();
             } else {
                 ee = classNode.getEnclosedElements();
             }
             EnumSet<ElementKind> elementKinds = getElementKind(result);
-            return ee.stream().filter(element -> elementKinds.contains(element.getKind())).collect(Collectors.toList());
+            List<Element> list = new ArrayList<>(elementKinds.size());
```

Comment:
i dont think elementKinds.size is predictive here

Reply:
yeah, wrong collection, should be `ee`